cleanlab_pinecone_rag_workflow/Cleanlab_Pinecone_RAG.ipynb

```diff
@@ -0,0 +1,2673 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "075f77c1-3328-4265-b077-160d496eb048",
+   "metadata": {},
+   "source": [
+    "# How to build a reliable, curated, and accurate RAG system using Cleanlab and Pinecone"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 125,
+   "id": "c70303b2-ce6f-4a03-a460-38bd27e8ae9b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "from sentence_transformers import SentenceTransformer\n",
+    "import time\n",
+    "from IPython.display import display\n",
+    "import warnings\n",
+    "import matplotlib.pyplot as plt\n",
+    "import os\n",
+    "import pinecone\n",
+    "import uuid\n",
+    "from pinecone import ServerlessSpec\n",
+    "from typing import List, Tuple, Dict, Optional\n",
+    "\n",
+    "warnings.simplefilter(action='ignore', category=FutureWarning)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 126,
+   "id": "c7ff0b6b-3a2b-4e6a-a41a-b0f7a54e32d2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "os.environ[\"PINECONE_API_KEY\"] = \"YOUR PINECONE API KEY HERE\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e7183c0f-c96f-4335-bc48-d474628dbe30",
+   "metadata": {
+    "id": "iMSW1p7td0yB"
+   },
+   "source": [
+    "Below is an example of how to set a Python index serverless specification, this allows us to define the cloud provider and region where we want to deploy our index. You can find a list of all [available providers and regions here](https://docs.pinecone.io/docs/projects)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 127,
+   "id": "28911e33-870c-4150-8bb8-e12f3352b9e2",
+   "metadata": {
+    "id": "KpRYFxYXd0yB"
+   },
+   "outputs": [],
+   "source": [
+    "cloud = os.environ.get('PINECONE_CLOUD') or 'aws'\n",
+    "region = os.environ.get('PINECONE_REGION') or 'us-east-1'\n",
+    "\n",
+    "spec = ServerlessSpec(cloud=cloud, region=region)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 128,
+   "id": "4e3d39bd-4726-4bf6-a016-aa0b136936f8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class PineconeRAGPipeline:\n",
+    "    def __init__(self, model_name: str = 'paraphrase-MiniLM-L6-v2', index_name: str = 'document-index', cloud: str = 'aws', region: str = 'us-east-1'):\n",
```

Comment:
nit but I think this is longer than black allows so I'd line break it (or whatever the linter suggests)